src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -58,6 +62,54 @@
 
     private SigningKeyResolver signingKeyResolver;
 
+    Claims expectedClaims = new DefaultClaims();
+
```

Comment:
I don't think this is necessary - we should invoke the delegate method immediately (like the others) and the delegate should have date-specific logic if necessary.

Users can call the delegate method with a custom claim name, and if that value is a Date, we still have to handle that correctly.